app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -437,6 +490,10 @@ def cluster_from_href(href, connection)
       connection.system_service.clusters_service.cluster_service(uuid_from_href(href)).get
     end
 
+    def storage_from_href(href, connection)
```

Comment:
We already fetch storage domain in different [place](https://github.com/ManageIQ/manageiq-providers-ovirt/blob/master/app/models/manageiq/providers/redhat/infra_manager/inventory/strategies/v4.rb#L72). I think that we start to have similar responsibilities in `inventory` and `ovirt_services`. In my opinion there is no need to have two distinct object having the same responsibility.

Reply:
we'd might access RHV from different places of the provider. We can think of introducing a class that will be responsible for interacting between the provider to RHV via v4, I'm not sure this is the PR to introduce such.

We should first get into agreement if we'd prefer that method over letting each workflow/scenario interact independently using the engine's connection.

Reply:
I am not saying that we should fix it in this PR. I am saying that we already have similar code in different place. I think it would be better to define responsibilities of the classes we use to talk to the engine.

Reply:
We'll track refactoring of ovirt_services and querying of engine's request by issue https://github.com/ManageIQ/manageiq-providers-ovirt/issues/100

Reply:
Thanks